format/FormatNexus.py

```diff
@@ -74,15 +74,15 @@ def _start(self):
                 len(reader.entries[0].instruments[0].detectors[0].modules) == 1
             ), "Currently only supports 1 NXdetector_module unless in a detector group"
 
+            self._raw_data = DataFactory(data, max_size=num_images)
             self._detector_model = DetectorFactory(
-                detector, self._beam_factory.model
+                detector, self._beam_factory.model, shape=self._raw_data.shape()
             ).model
-            self._raw_data = DataFactory(data, max_size=num_images)
         else:
+            self._raw_data = detectorgroupdatafactory(data, instrument)
             self._detector_model = DetectorFactoryFromGroup(
```

Comment:
@phyy-nx you'll note here that the shape cannot be passed to DetectorFactoryFromGroup so this only applies to single panel detectors. `self._raw_data` assignment moved for consistency.